src/main/webapp/app/exercises/shared/result/result.utils.ts

```diff
@@ -288,3 +290,66 @@ export const isBuildFailed = (submission?: Submission) => {
 export const isManualResult = (result?: Result) => {
     return result?.assessmentType !== AssessmentType.AUTOMATIC;
 };
+
+/**
+ * Retrieves a list of test cases names contained in a result's feedback list.
+ *
+ * @param results list of results to extract the test case names from
+ * @return list of extracted test case names
+ * @private
+ */
+export function getTestCaseNamesFromResults(results: ResultWithPointsPerGradingCriterion[]): string[] {
+    const testCasesNames: Set<string> = new Set();
+    results.forEach((result) => {
+        if (!result.result.feedbacks) {
+            return [];
+        }
+        result.result.feedbacks.map((f) => {
+            if (f.type && [FeedbackType.AUTOMATIC, FeedbackType.AUTOMATIC_ADAPTED].includes(f.type)) {
```

Comment:
We have `isTestCaseFeedback` in `feedback.modelt.ts`.
That should also do the work

Reply:
Nice suggestion thanks 🙂